R/travis-auth.R

```diff
@@ -1,75 +1,69 @@
-auth_travis_ <- function(gh_token = NULL) {
-  cli::cat_bullet(
-    bullet = "pointer", bullet_col = "yellow",
-    " Authenticating to GitHub."
-  )
-  if (is.null(gh_token)) {
-    # Do not allow caching this token, it needs to be fresh
-    gh_token <- auth_github_(
-      "read:org", "user:email", "repo_deployment", "repo:status",
-      "read:repo_hook", "write:repo_hook"
+auth_travis <- function(endpoint = ".org") {
+  yml <- tryCatch({
+    readLines("~/.travis/config.yml")
+  },
+  warning = function(cond) {
```

Comment:
This looks mildly dangerous. Can we just stop here and ask the user to call a function that creates the file? Do we need to store in `~/.travis` or can we use {rappdirs}?

Reply:
What might be dangerous about it?

`rappdirs` would certainly be the most robust approach. Though this file was not chosen arbitrary since it might already exists if the users interacted with the travis CLI beforehand. Probably _rappdirs_ would change the dirs?

Reply:
I don't think we should write a config file as a side effect in case of a warning. If it's a one-time procedure, it's fine check if the file exists, and if not, err with clear instructions what to do, e.g., call a function exported by {travis}.